### PR TITLE
DCOS-23834 - Data Layer Reference Implementation

### PR DESCRIPTION
**Pull request by Fabs**
Being this a feature branch, we already reviewed the code in separate. If you still feel there are code improvements I can look into it, but the size of the PR makes it harder to code review the code this second time.

Steps to test:
- Run it and check the Packages Repositories
- Test adding, removing, filtering packages. Remember to save the link to the universe before removing it.
- You could also check the requests being fired on the Network tab to see if it is all good.

NOTE:
We still missing 2 or 3 improvements on this ticket, specially typescript, more graphql and etc. but I believe we achieve the goal to remove the Mixins and the stores already, and I dont want this code to get super old without hitting master.

### CODE REVIEW

**Comment by nLight** [comment]
Flicts!

**Comment by Fabs** [comment]
the tests are green. What fails is the last steps related to release:
image

**Comment by Fabs** [comment]
if @juliangieseke or @DanielMSchmidt could give me some direction on how to fix that, it would be cool.

File: Jenkinsfile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 @Library("sec_ci_libs@v2-latest") _
 
-def master_branches = ["master", ] as String[]
```
Comment (juliangieseke) [inline Jenkinsfile]: can you please remove your branch from Jenkinsfile once your done? 👍

**Comment by juliangieseke** [comment]
fixed Jenkinsfile and rebased onto latest master 👍

**Comment by juliangieseke** [comment]
> ERROR 2018-05-04 15:51:15 system:  ✘ Universe installs a community package
Legit Fail?

**Comment by Fabs** [comment]
@juliangiesek. The system test could be broken. I will check Monday. Thanks for putting extra effort on this one.

**Comment by juliangieseke** [comment]
Hm, now its green 👍

**Comment by DanielMSchmidt** [comment]
@Fabs Conflicts in the shrinkwrap

**Comment by Fabs** [comment]
Going to fix that today.

**Comment by Fabs** [comment]
@juliangieseke and @DanielMSchmidt . Fixed it, and tests are green YAY

**Comment by Fabs** [comment]
@DanielMSchmidt, @juliangieseke  and maybe even @nLight. That would be awesome if we could try to get this in before we merge master again and this gets out of sync.

**Comment by Fabs** [comment]
@juliangieseke I give you a mesosphere fritz kola for everytime we find a bug here :-)

**Pull request merged**